bot/decorators.py

```diff
@@ -236,3 +238,31 @@ async def wrapped(*args, **kwargs) -> t.Any:
             return await func(*args, **kwargs)
         return wrapped
     return decorator
+
+
+def ensure_duration_in_future(duration_arg: function.Argument) -> t.Callable:
+    """
+    Ensure the duration argument is in the future.
+
+    If the condition fails, a warning is sent to the invoking context.
+
+    `duration_arg` is the keyword name or position index of the parameter of the decorated command
+    whose value is the target duration.
+
+    This decorator must go before (below) the `command` decorator.
+    """
+    def decorator(func: types.FunctionType) -> types.FunctionType:
+        @command_wraps(func)
+        async def wrapper(*args, **kwargs) -> t.Any:
+            bound_args = function.get_bound_args(func, args, kwargs)
+            target = function.get_arg_value(duration_arg, bound_args)
+
+            ctx = function.get_arg_value(1, bound_args)
+
+            if isinstance(target, datetime) and target < arrow.utcnow():
```

Comment:
Subtle thing but using less than instead of less than equal is misleading when it's supposed to ensure it's in the _future_.

Reply:
resolved in 61feabb